File: convalida-validators/src/main/java/convalida/validators/NotEmptyValidator.java
package convalida.validators;

import android.widget.EditText;

/**
 * @author Wellington Costa on 21/06/2017.
 */
public class NotEmptyValidator extends AbstractValidator {

    public NotEmptyValidator(
            EditText editText,
            String errorMessage,
            boolean autoDismiss) {
        super(editText, errorMessage, autoDismiss);
    }

    @Override public boolean isNotValid(String value) {
        return value.isEmpty();
    }

}

File: convalida-validators/src/main/java/convalida/validators/NotEmptyValidator.kt
package convalida.validators

import android.widget.EditText

/**
 * @author Wellington Costa on 21/06/2017.
 */
class NotEmptyValidator(editText: EditText, errorMessage: String) : AbstractValidator(editText, errorMessage) {

    override fun isNotValid(value: String): Boolean {
        return value.isEmpty()
    }

}